langchain/src/embeddings/ollama.ts

```diff
@@ -0,0 +1,134 @@
+import { OllamaInput, OllamaRequestParams } from "../util/ollama.js";
+import { Embeddings, EmbeddingsParams } from "./base.js";
+
+type CamelCasedRequestOptions = Omit<OllamaInput, "baseUrl" | "model">;
+
+/**
+ * Interface for OllamaEmbeddings parameters. Extends EmbeddingsParams and
+ * defines additional parameters specific to the OllamaEmbeddings class.
+ */
+interface OllamaEmbeddingsParams extends EmbeddingsParams {
+  /** The Ollama model to use, e.g: "llama2:13b" */
+  model: string;
+
+  /** Base URL of the Ollama server, defaults to "http://localhost:11434" */
+  url?: string;
+
+  /** Advanced Ollama API request parameters in camelCase, see
+   * https://github.com/jmorganca/ollama/blob/main/docs/modelfile.md#valid-parameters-and-values
+   * for details of the available parameters.
+   */
+  requestOptions?: CamelCasedRequestOptions;
+}
+
+export class OllamaEmbeddings extends Embeddings {
+  model: string;
+
+  baseUrl = "http://localhost:11434";
+
+  requestOptions?: OllamaRequestParams["options"];
+
+  constructor(params: OllamaEmbeddingsParams) {
+    super(params);
+
+    this.model = params.model;
+
+    if (params.url) {
+      this.baseUrl = params.url;
+    }
+
+    if (params.requestOptions) {
+      this.requestOptions = this._convertOptions(params.requestOptions);
+    }
+  }
+
+  /** convert camelCased Ollama request options like "useMMap" to
+   * the snake_cased equivalent which the ollama API actually uses.
+   * Used only for consistency with the llms/Ollama and chatModels/Ollama classes
+   */
+  _convertOptions(requestOptions?: CamelCasedRequestOptions) {
+    if (!requestOptions) {
+      return;
```

Comment:
Might suggest throwing here instead or defaulting to an empty object?

Reply:
not necessary anymore, since the function is only called when requestOptions are set and the argument is not optional anymore after refactoring.